specification.markdown

```diff
@@ -228,6 +210,95 @@ VINT\_WIDTH | VINT\_MARKER | VINT\_DATA     | Element Data Size Status
         For example -2 can be coded as 0xFFFFFFFFFFFFFE or 0xFFFE or
         0xFE and 5 can be coded 0x000000000005 or 0x0005 or 0x05.
 
+## EBML Element Types
+
+Each defined EBML Element MUST have a declared Element Type. The Element Type defines a concept for storing data that may be constrained by length, endianness, and purpose.
+
+Element Data Type:   Signed Integer
+
+    Endianness:     Big-endian
+    Length:         A Signed Integer Element MUST at least 1 octet in
```

Comment:
Here are some notes to ensure that I haveve understood this conversation correctly.

- Consensus now is that 0 length data will be allowed for unsigned and signed integers.

- Shall we then also allow 0 for floats with a meaning of 0?

- Zero length dates seem a bit ridiculous. I suggest not allowing a zero length date.

- A zero length element shall indicate a meaning declared in the 'data type' section of the ebml spec. A zero length element is not intended to be interpreted to the default value of that element (the default is used for elements that are missing AND mandatory).

- Whether or not an element uses a default is defined in the semantics rather than here in the definitions of data types.